airflow/sensors/external_task.py

```diff
@@ -163,9 +163,11 @@ def __init__(
         total_states = set(self.allowed_states + self.skipped_states + self.failed_states)
 
         if len(total_states) != len(self.allowed_states) + len(self.skipped_states) + len(self.failed_states):
-            raise AirflowException(
-                "Duplicate values provided across allowed_states, skipped_states and failed_states."
-            )
+            # TODO: remove this if block when min_airflow_version is set to higher than 2.7.1
+            message = "Duplicate values provided across allowed_states, skipped_states and failed_states."
+            if self.soft_fail:
+                raise AirflowSkipException(message)
+            raise AirflowException(message)
```

Comment:
If there is a problem in the init, it will never arrive to an execution state, so don't need it.

Reply:
Right, I'll change it. Thanks :)

Reply:
@hussein-awala Right, I'll change it. Thanks for pointing it out. :)

Just wanted to confirm my understanding, if there is a problem in the `__init__` it will be caught in the parsing phase and because of that, we don't need the `AirflowSkipException` exception, right?

Reply:
In addition to that, I think https://github.com/apache/airflow/pull/33424 also handles it and that's why we have the TODO in providers but not in the core.

Reply:
Thanks, @Lee-W I'll remove the ToDo.

Reply:
Sorry, what I meant was the whole section is not needed instead of just the TODO

Reply:
https://github.com/apache/airflow/blob/19450e03f534f63399bf5db2df7690fdd47b09c8/airflow/sensors/base.py#L242-L259

it's now handled in the base sensor

Reply:
Thanks for clarification,  I'll update the PR and will only retain the meaningful tests.